test/e2e/specs/site-editor/template-part.spec.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * WordPress dependencies
+ */
+const {
+	test,
+	expect,
+	Editor,
+} = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	editor: async ( { page }, use ) => {
+		await use( new Editor( { page, hasIframe: true } ) );
+	},
+} );
+
+test.describe( 'Template Part', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'can create template parts via the block placeholder start blank option', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'emptytheme//header',
+			postType: 'wp_template_part',
+		} );
+
+		// Insert a new template block and 'start blank'.
+		await editor.insertBlock( { name: 'core/template-part' } );
+		await editor.canvas.click( 'role=button[name="Start blank"i]' );
+
+		// Fill in a name in the dialog that pops up.
+		await page.type( 'role=dialog >> role=textbox[name="Name"i]', 'New' );
+		await page.keyboard.press( 'Enter' );
+
+		// The template part should be visible with a block appender.
+		const templatePart = editor.canvas.locator(
+			'div[data-type="core/template-part"]'
+		);
+		const addBlockButton = templatePart.locator(
+			'role=button[name="Add block"i]'
+		);
+		await expect( templatePart ).toBeVisible();
+		await expect( addBlockButton ).toBeVisible();
+	} );
+
+	test( 'can create template parts via the block placeholder choose existing option', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		// Visit the index.
+		await admin.visitSiteEditor();
+
+		const headerTemplateParts = editor.canvas.locator(
+			'[data-type="core/template-part"]'
+		);
+
+		// There should be 1 template part already in the index template.
+		await expect( headerTemplateParts ).toHaveCount( 1 );
+
+		// Insert a new template block and choose an existing header pattern.
+		await editor.insertBlock( { name: 'core/template-part' } );
+		await editor.canvas.click( 'role=button[name="Choose"i]' );
+		await page.click(
+			'role=listbox[name="Block Patterns"i] >> role=option >> text="header"'
```

Comment:
Why can't we just do this?

```suggestion
			'role=listbox[name="Block Patterns"i] >> role=option[name="header"i]'
```